src/Arcus.EventGrid.Security/EventGridSubscriptionValidator.cs

```diff
@@ -53,7 +55,7 @@ private async Task<HttpResponseMessage> HandleSubscriptionValidationEventAsync(H
         {
             // Parsing the incoming message to a typed EventGrid message
             var rawRequest = await actionContext.Request.Content.ReadAsStringAsync();
-            var gridMessage = EventGridParser.Parse<SubscriptionValidation>(rawRequest);
+            var gridMessage = EventGridParser.Parse<EventGridEvent<SubscriptionValidationEventData>>(rawRequest);
```

Comment:
Do we have to specify `EventGridEvent<TData>` here? Isn't `TData` enough?

Reply:
Hmm, not if we want backwares capability I guess. If we don't have specific events anymore; we could indeed refactor this to only pass in the data type.

Reply:
We don't need backwards compatibility given this is for a new major version.

Think I've mentioned this before, no? 😅

Reply:
Hmm, that doesn't make sense at all to me. I thought t the whole idea was to make the transistion graceful and make every event and event operation compatible with both Azure event and our own.

Reply:
This will be released as part of v3.0 which means we will break stuff, we do not have to be compatible with v2.x.

That said, our new approach should work with custom event payloads, raw payloads & Azure payloads.

Reply:
Ah, now I see it. The `SubscriptionValidationEventData` is a type from the Microsoft SDK which means that without an event implementation on that data type; we should use the general `EventGridEvent<>` class.

So to conclude: no `TData` is not enough here.

Reply:
I'm not sure if I'm following - Why does it matter if it's one of our events or one provided by MSFT?
It feels like duplication, without added value, to specify `EventGridEvent` every time given we are already using `EventGridParser`

Reply:
Than we should provide another parsing function where you could specify the data only since otherwise custom event implementations wouldn't work.

Reply:
I think we'll have to discuss this in person but we should only provide one parsing function, regardless of whom created the event/event data.

Reply:
Ok, added overload(s) `Parse(FromData)<TEventData>` with and without specified session ID to parse events only by their event data payload type while still supporting custom event implementations.